virttest/env_process.py

```diff
@@ -129,6 +129,9 @@ def preprocess_vm(test, params, env, name):
         pass
     if create_vm:
         vm = env.create_vm(vm_type, target, name, params, test.bindir)
+    elif not isinstance(vm, virt_vm.BaseVM.lookup_vm_class(vm_type, target)):
+        # Force-recreate when vm-class changes
+        vm = env.create_vm(vm_type, target, name, params, test.bindir)
```

Comment:
Suggest:
```python
if not vm:
    create_vm = True
elif vm_type == 'libvirt':
    connect_uri = libvirt_vm.normalize_connect_uri(connect_uri)
    if (not vm.connect_uri == connect_uri):
        create_vm = True
elif not isinstance(vm, virt_vm.BaseVM.lookup_vm_class(vm_type, target)):
    # Force-recreate when vm-class changes
    create_vm = True

if create_vm:
    vm = env.create_vm(vm_type, target, name, params, test.bindir)
```

Reply:
that would not cover the situation where `vm_type` was `qemu` and changed to `libvirt` and that would actually crash because `qemu_vm.VM.connect_uri` does not exists. Updating so it copes with that as well...